skimage/feature/tests/test_peak.py

```diff
@@ -85,6 +85,20 @@ def test_num_peaks():
     assert (3, 5) in peaks_limited
 
 
+def test_num_peaks_and_labels():
+    image = np.zeros((7, 7), dtype=np.uint8)
+    labels = np.zeros((7, 7), dtype=np.uint8) + 20
+    image[1, 1] = 10
+    image[1, 3] = 11
+    image[1, 5] = 12
+    image[3, 5] = 8
+    image[5, 3] = 7
+    assert len(peak.peak_local_max(image, min_distance=1, threshold_abs=0)) == 5
+    peaks_limited = peak.peak_local_max(
+        image, min_distance=1, threshold_abs=0, num_peaks=2)
+    assert len(peaks_limited) == 2
```

Comment:
This test does not seem to address the original issue. To cover the case, we need at least 2 labels for input image and call `peak_local_max` with `labels` parameter.